### PR TITLE
[#27] [UI] As a user, I can see the dropdown text in the Question page

### LINKED ISSUE #27: [UI] As a user, I can see the dropdown text in the Question page
## Why
The user can see the questions and the corresponding answers in the Question page so he can answer them.

## Acceptance Criteria
- Create the dropdown text component.
[Figma](https://www.figma.com/file/i1AI0IzCWuJApX5CK0DmiW/Challenge---Web-App?type=design&node-id=681-474&t=apcSQ1nCik2OLC0u-0)
image

### PR DESCRIPTION
**Pull request by manh-t**
Close #27 

## What happened 👀

Create the dropdown component to prepare for submitting the survey.

## Insight 📝

We don't have any dropdown item's design so I create one on my own.

## Proof Of Work 📹

image

### CODE REVIEW

**Comment by manh-t** [comment]
🚀 Deployed on https://64f93c1e4a564625014d70dd--manh-react-survey.netlify.app

**Comment by manh-t** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Congrats, coverage is good</td>
    </tr>
  </tbody>
</table>

## Coverage in All Files
File | Line Coverage | Statement Coverage | Function Coverage | Branch Coverage
---- | ------------: | -----------------: | ----------------: | --------------:
[src/types/question.ts](../blob/8d38cea/src/types/question.ts) | (1/2) 50% | (1/2) 50% | (0/1) 0% | (0/2) 0%
[src/../TextArea/index.tsx](../blob/8d38cea/src/components/TextArea/index.tsx) | (5/8) 63% | (5/8) 63% | (1/2) 50% | (0/2) 0%
[src/../AppSlider/index.tsx](../blob/8d38cea/src/components/AppSlider/index.tsx) | (4/6) 67% | (4/6) 67% | (1/3) 33% | (3/3) 100%
[src/tests/error.ts](../blob/8d38cea/src/tests/error.ts) | (2/3) 67% | (2/3) 67% | (1/2) 50% | (1/3) 33%
[src/../Rating/index.tsx](../blob/8d38cea/src/components/Rating/index.tsx) | (18/22) 82% | (18/22) 82% | (7/7) 100% | (4/13) 31%
[src/../Content/index.tsx](../blob/8d38cea/src/components/Dashboard/Content/index.tsx) | (14/15) 93% | (14/15) 93% | (7/7) 100% | (6/9) 67%
[src/helpers/deserializer.ts](../blob/8d38cea/src/helpers/deserializer.ts) | (17/18) 94% | (17/18) 94% | (7/7) 100% | (10/12) 83%
[src/../v1/requestManager.ts](../blob/8d38cea/src/lib/request/v1/requestManager.ts) | (19/20) 95% | (19/20) 95% | (6/6) 100% | (3/5) 60%
[src/App.tsx](../blob/8d38cea/src/App.tsx) | (3/3) 100% | (3/3) 100% | (1/1) 100% | (0/0) NaN%
[src/index.tsx](../blob/8d38cea/src/index.tsx) | (3/3) 100% | (3/3) 100% | (0/0) NaN% | (0/0) NaN%
[src/../Authentication/index.ts](../blob/8d38cea/src/adapters/Authentication/index.ts) | (4/4) 100% | (4/4) 100% | (2/2) 100% | (0/0) NaN%
[src/adapters/Base/index.ts](../blob/8d38cea/src/adapters/Base/index.ts) | (8/8) 100% | (8/8) 100% | (2/2) 100% | (2/2) 100%
[src/adapters/BaseAuth/index.ts](../blob/8d38cea/src/adapters/BaseAuth/index.ts) | (10/10) 100% | (10/10) 100% | (4/4) 100% | (2/2) 100%
[src/adapters/Survey/index.ts](../blob/8d38cea/src/adapters/Survey/index.ts) | (2/2) 100% | (2/2) 100% | (2/2) 100% | (0/0) NaN%
[src/adapters/User/index.ts](../blob/8d38cea/src/adapters/User/index.ts) | (1/1) 100% | (1/1) 100% | (1/1) 100% | (0/0) NaN%
Other (39 more) | (263/263) 100% | (263/263) 100% | (95/96) 99% | (80/85) 94%
Total | (374/388) 96% | (374/388) 96% | (137/143) 96% | (111/138) 80%

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 440f8ac79caaf0b21303957047ce1d0cca59ce6f
</p>

File: src/components/Dropdown/index.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  questionId: string;
+  items: Answer[];
+  onValueChanged?: (value: Answer) => void;
+}
+const Dropdown = ({ questionId, items, onValueChanged }: DropdownProps): JSX.Element => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectedValue, setSelectedValue] = useState(items[0].text);
+
+  const toggleDropdown = () => {
+    setIsOpen((prev) => !prev);
```
Comment (nvminhtue) [inline src/components/Dropdown/index.tsx]:
```suggestion
    setIsOpen(!isOpen);
```

File: src/components/Dropdown/index.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  };
+
+  const handleOnSelectValue = (value: Answer) => {
+    setSelectedValue(value.text);
+    onValueChanged?.(value);
```
Comment (nvminhtue) [inline src/components/Dropdown/index.tsx]: this `onValueChanged` is making no sense now, in what case we don't pass the `onValueChanged` prop into this component?

File: src/components/Dropdown/index.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  };
+
+  useEffect(() => {
+    setSelectedValue(items[0].text);
+  }, [questionId, items]);
+
+  return (
+    <div className="relative flex flex-col items-center w-full rounded-lg" data-test-id={dropdownDataTestIds.base}>
+      <button
+        onClick={() => toggleDropdown()}
```
Comment (nvminhtue) [inline src/components/Dropdown/index.tsx]:
```suggestion
        onClick={toggleDropdown}
```

File: src/components/Dropdown/index.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        className="w-full h-14 flex items-center justify-between appearance-none bg-white bg-opacity-[.18] rounded-[12px] hover:outline-none hover:ring-transparent px-3 text-white text-regular tracking-survey-tight hover:bg-opacity-30"
+      >
+        {selectedValue}
+        {!isOpen ? <ArrowDropdown /> : <ArrowDropdown className="rotate-180" />}
```
Comment (nvminhtue) [inline src/components/Dropdown/index.tsx]:
```suggestion
        <ArrowDropdown className={isOpen ? '' : 'rotate-180'} />
```

File: src/components/Dropdown/index.tsx (hunk 2 above)
Comment (manh-t) [reply src/components/Dropdown/index.tsx]: @nvminhtue I think we always pass `onValueChanged` to this component 👍 Let me update to make `onValueChanged` as a mandatory prop

File: src/components/Dropdown/index.tsx (hunk 1 above)
Comment (manh-t) [reply src/components/Dropdown/index.tsx]: @nvminhtue fixed in [4803f51](https://github.com/manh-t/react-survey/pull/78/commits/4803f511251c858a44008f8f4f3580956f568ca8)

File: src/components/Dropdown/index.tsx (hunk 2 above)
Comment (manh-t) [reply src/components/Dropdown/index.tsx]: @nvminhtue fixed in [4803f51](https://github.com/manh-t/react-survey/pull/78/commits/4803f511251c858a44008f8f4f3580956f568ca8)

File: src/components/Dropdown/index.tsx (hunk 4 above)
Comment (manh-t) [reply src/components/Dropdown/index.tsx]: @nvminhtue fixed in [4803f51](https://github.com/manh-t/react-survey/pull/78/commits/4803f511251c858a44008f8f4f3580956f568ca8)

File: src/components/Dropdown/index.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    onValueChanged(value);
+    toggleDropdown();
+  };
+
+  useEffect(() => {
+    setSelectedValue(items[0].text);
+  }, [questionId, items]);
+
+  return (
+    <div className="relative flex flex-col items-center w-full rounded-lg" data-test-id={dropdownDataTestIds.base}>
```
Comment (hoangmirs) [inline src/components/Dropdown/index.tsx:34]: Usually, we just use the [`select`](https://www.w3schools.com/tags/tag_select.asp) for the dropdown if there is no design. But building it with React is not that hard, right? 😄

**Review by hoangmirs — APPROVED** [review]

File: src/components/Dropdown/index.tsx (hunk 5 above)
Comment (manh-t) [reply src/components/Dropdown/index.tsx:34]: @hoangmirs yeah, I'm trying to create a custom component on my own and the result is not bad 🚀

**Review by nvminhtue — APPROVED** [review]

**Pull request merged**